packages/api/src/utils.js

```diff
@@ -93,4 +93,35 @@ const decodeStateTransition = async (client, base64) => {
   return decoded
 }
 
-module.exports = { hash, decodeStateTransition, getKnex }
+const calculateInterval = (start, end) => {
+  const intervals = {
+    PT5M: 300000,
+    PT30M: 1800000,
+    PT1H: 3600000,
+    PT2H: 7200000,
+    PT12H: 43200000,
+    P1D: 86400000,
+    P1W: 604800000,
+    P1M: 2419200000,
+    P1Y: 29030400000
+  }
+
+  const intervalsInRFC = Object.keys(intervals)
+
+  const startTimestamp = start.getTime()
+  const endTimestamp = end.getTime()
+
+  const period = endTimestamp - startTimestamp
+
+  for (let i = 0; i < intervalsInRFC.length; i++) {
```

Comment:
How its working? maybe rewrite in the functional way? not understandable

Reply:
Has been upgraded to a new one.
The point of this loop is to find such an interval value that the number of periods on the graph is not less than 2, but at the same time it is not too big either